e_commerce_app/HomeView/HomeViewModel.swift

```diff
@@ -5,36 +5,66 @@
 //  Created by Bhavin Kapadia on 2022-09-13.
 //
 
-import CoreData
 import UIKit
 
+enum ResponseError: Error {
+    case unknownAPIResponse
+    case generic
+}
+
 final class HomeViewModel {
-    let userID: UUID
+    var products: [Product] = []
     
+    let userID: UUID
```

Comment:
We already talked about this, but if you have time you could initialize with the user instead of the ID. This just negates having to fetch the user all the time. Once again, you could treat this as a clean up at the end after you have everything working.